kernel/src/lib.rs

```diff
@@ -634,15 +638,18 @@ pub trait Engine: AsAny {
 }
 
 // we have an 'internal' feature flag: default-engine-base, which is actually just the shared
-// pieces of default-engine and default-engine-rustls. the crate can't compile with _only_
+// pieces of default-engine-native-tls and default-engine-rustls. the crate can't compile with _only_
 // default-engine-base, so we give a friendly error here.
 #[cfg(all(
     feature = "default-engine-base",
-    not(any(feature = "default-engine", feature = "default-engine-rustls",))
+    not(any(
+        feature = "default-engine-native-tls",
+        feature = "default-engine-rustls",
+    ))
 ))]
 compile_error!(
     "The default-engine-base feature flag is not meant to be used directly. \
-    Please use either default-engine or default-engine-rustls."
+    Please use either default-engine-native-tls or default-engine-rustls."
```

Comment:
Typically, the correct thing to do is to use native-tls right? If so, can we give a recommendation here? 
```suggestion
    Please use either default-engine-native-tls or default-engine-rustls. default-engine-native-tls is recommended for most use cases."
```

Reply:
@OussamaSaoudi native-tls brings in `openssl-sys` which adds a hidden build-time dependency of `libssl-dev` (or equivalent), and also makes cross-compiling (e.g. for lambda, dotnet) perilous at best.

IMHO in terms of quick compile times, safe code, and good vibes, rustls should be used by default unless somebody needs openssl's behaviors